### PR TITLE
Fix AVIF image decoding

### PR DESCRIPTION
**Pull request by brianreavis**
# Objective / Solution

This expands upon:

- https://github.com/bevyengine/bevy/pull/15586

The image crate requires `avif-native` feature for decoding AVIF images ([documented here](https://github.com/image-rs/image?tab=readme-ov-file#supported-image-formats)). The `avif` flag only allows for the encoding of AVIFs.

## Testing

- Tested on a local project

### CODE REVIEW

File: crates/bevy_internal/Cargo.toml [hunk 1]
```diff
@@ -45,7 +45,7 @@ zlib = ["bevy_image/zlib"]
 zstd = ["bevy_image/zstd"]
 
 # Image format support (PNG enabled by default)
-avif = ["bevy_image/avif"]
+avif = ["bevy_image/avif-native"]
```
Comment (alice-i-cecile) [inline crates/bevy_internal/Cargo.toml:48]:
```suggestion
avif = ["bevy_image/avif"]
```
We need to enable the `bevy_image` feature flag here, which is still just called `avif`.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
The feature flag for `bevy_image` is wrong :)

**Comment by alice-i-cecile** [comment]
@clarfonthey can I get your review here?

**Comment by brianreavis** [comment]
@alice-i-cecile You're quick! Just noticed that too and fixed it.

**Comment by brianreavis** [comment]
One thing that needs to be figured out: the `ci` clippy job fails because it runs with `--all-features`:
https://github.com/bevyengine/bevy/actions/runs/11371344253/job/31633201423?pr=15959#step:6:876

We could install `dav1d`, but that seems overkill to make that particular job work.

**Comment by mockersf** [comment]
AVIF seems to be mostly a format for animated images, which Bevy doesn't support built in.
I think the fix is to not expose it in Bevy, it will still be possible for a third party plugin to enable it.

**Comment by brianreavis** [comment]
>  AVIF seems to be mostly a format for animated images, which Bevy doesn't support built in.

Though AVIF supports animation, it's great for compressing single images: https://jakearchibald.com/2020/avif-has-landed/

**Review by clarfonthey — APPROVED** [review]
I'm fine with this; it does look like the decoder is only gated on this feature, not the `avif` feature.

**Comment by mockersf** [comment]
> the `ci` clippy job fails because it runs with `--all-features`

You can either remove built-in support for `AVIF`, which is what I would recommend, or add the `dav1d` library to the linux dependencies ([for ci](https://github.com/bevyengine/bevy/blob/main/.github/actions/install-linux-deps/action.yml) and [in documentation](https://github.com/bevyengine/bevy-website/blob/main/content/learn/book/getting-started/installation.md)).

Both options are possible, it depends how much you want built-in support for `AVIF`

**Comment by clarfonthey** [comment]
I feel like it'd be best as a third option, opt for an embedded AVIF implementation, but in the absence of that, removing it feels better.

**Comment by alice-i-cecile** [comment]
> I feel like it'd be best as a third option, opt for an embedded AVIF implementation, but in the absence of that, removing it feels better.

I don't think this is an important enough feature for that. I'd prefer to remove it and let third-party solutions cover this.

**Comment by brianreavis** [comment]
I’m going to go ahead and close this PR and leave removal to another PR. If I get time I’ll take a stab at it.

**Pull request closed without merging**